internal/client/clustergroup/iam_policy/iam_resource.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package iamclustergroupclient
+
+import (
+	"fmt"
+
+	"github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/client/transport"
+	clustergroupmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/clustergroup"
+	iammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy"
+	clustergroupiammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy/clustergroup"
+)
+
+// New creates a new cluster group IAM policy resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+  Client for cluster group IAM policy resource service API.
+*/
+
+type Client struct {
+	*transport.Client
+}
+
+// ClientService is the interface for Client methods.
+type ClientService interface {
+	ManageV1alpha1ClusterGroupIAMPolicyGet(fn *clustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupFullName) (*clustergroupiammodel.VmwareTanzuManageV1alpha1ClustergroupGetClusterGroupIAMPolicyResponse, error)
+
+	ManageV1alpha1ClusterGroupIAMPolicyPatch(request *clustergroupiammodel.VmwareTanzuManageV1alpha1ClustergroupPatchClusterGroupIAMPolicyRequest) (*clustergroupiammodel.VmwareTanzuManageV1alpha1ClustergroupPatchClusterGroupIAMPolicyResponse, error)
+
+	ManageV1alpha1ClusterGroupIAMPolicyUpdate(fn *clustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupFullName, request *iammodel.VmwareTanzuCoreV1alpha1PolicyIAMPolicy) (*clustergroupiammodel.VmwareTanzuManageV1alpha1ClustergroupUpdateClusterGroupIAMPolicyResponse, error)
+}
+
+/*
+  ManageV1alpha1ClusterGroupIAMPolicyGet gets policy for a cluster group.
+*/
+
+func (c *Client) ManageV1alpha1ClusterGroupIAMPolicyGet(fn *clustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupFullName) (*clustergroupiammodel.VmwareTanzuManageV1alpha1ClustergroupGetClusterGroupIAMPolicyResponse, error) {
+	requestURL := fmt.Sprintf("%s/%s", "v1alpha1/clustergroups:iam", fn.Name)
```

Comment:
make it a constant "v1alpha1/clustergroups:iam" and similar changes in other places as well